pkg/storage/chunk/cache/instrumented.go

```diff
@@ -116,3 +122,14 @@ func (i *instrumentedCache) Fetch(ctx context.Context, keys []string) ([]string,
 func (i *instrumentedCache) Stop() {
 	i.Cache.Stop()
 }
+
+func (i *instrumentedCache) getTenantID(ctx context.Context) string {
+	// if per-tenant metrics are disabled, set an empty label which will be ignored
+	if !i.perTenant {
+		return ""
+	}
+
+	// best effort: if tenant ID cannot be retrieved, set an empty label which will be ignored
+	userID, _ := tenant.TenantID(ctx)
```

Comment:
double checking: don't you think it would be worth it to log the error returned from the tenantID() call here?

Reply:
It's a good question. I ignored the error here because operationally it's not of much use, and if we can't derive the tenant from the context then we'll have quite a few upstream errors to contend with.

Reply:
got it, thanks!